dds/DCPS/transport/shmem/ShmemReceiveStrategy.cpp

```diff
@@ -43,14 +43,18 @@ ShmemReceiveStrategy::read()
     bound_name_ = "Write-" + link_->local_address();
   }
 
-  ShmemAllocator* alloc = link_->peer_allocator();
   void* mem = 0;
-  if (alloc == 0 || -1 == alloc->find(bound_name_.c_str(), mem)) {
-    VDBG_LVL((LM_DEBUG, "(%P|%t) ShmemReceiveStrategy::read link %@ "
-              "peer allocator not found, receive_bytes will close link\n",
-              link_), 1);
-    handle_dds_input(ACE_INVALID_HANDLE); // will return 0 to the TRecvStrateg.
-    return;
```

Comment:
This looks like a data race on `alloc` object in master?

Reply:
> This looks like a data race on `alloc` object in master?

Correct.  That's why I introduced the `PeerAllocatorProxy`.